packages/web-components/src/components/leaving-ibm/leaving-ibm-modal.ts

```diff
@@ -22,6 +22,16 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-leaving-ibm-modal`)
 class DDSLeavingIbmModal extends StableSelectorMixin(BXModal) {
+  connectedCallback() {
+    if (!this.hasAttribute('role')) {
+      this.setAttribute('role', 'dialog');
+    }
+    if (!this.hasAttribute('aria-labelledby')) {
+      this.setAttribute('aria-labelledby', 'leaving-ibm-heading');
+    }
+    super.connectedCallback();
+  }
```

Comment:
To avoid (theoretical) ID duplication (though it's less likely for an app to have more than one leaving IBM modal), we can auto-generate the ID, like below.`@HostListener` should be from `import HostListener from 'carbon-web-components/es/globals/decorators/host-listener.js'`:

```suggestion
  /**
   * The unique ID for ID ref.
   */
  private _uniqueId = `__${ddsPrefix}-leaving-ibm-modal__${Math.random()
    .toString(36)
    .slice(2)}`;

  /**
   * Handles `slotchange` event.
   *
   * @param event The event.
   */
  @HostListener('shadowRoot:slotchange')
  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
  private _handleSlotChange = (event: Event) => {
    const { selectorHeading } = this.constructor as typeof DDSLeavingIbmModal;
    if (!this.hasAttribute('aria-labelledby')) {
      const headingNode = (event.target as HTMLSlotElement).assignedNodes().reduce((acc, node) => {
        if ((node as Element).matches?.(selectorHeading)) {
          acc.push(node);
        } else {
          acc.push(...((node as Element).querySelectorAll?.(selectorHeading) ?? []));
        }
        return acc;
      }, [] as Node[])[0] as Element | void;
      if (headingNode) {
        if (!headingNode.id) {
          const { id, _uniqueId: uniqueId } = this;
          headingNode.id = id || uniqueId;
        }
        if (headingNode.id) {
          this.setAttribute('aria-labelledby', headingNode.id);
        }
      }
    }
  };

  connectedCallback() {
    if (!this.hasAttribute('role')) {
      this.setAttribute('role', 'dialog');
    }
    super.connectedCallback();
  }

  /**
   * A selector selecting the heading.
   */
  static get selectorHeading() {
    return `${ddsPrefix}-leaving-ibm-modal-heading`;
  }
```

Reply:
oh good point! Thanks @asudoh!